graph/appender/dead_node.go

```diff
@@ -37,7 +85,7 @@ func applyDeadNodes(trafficMap graph.TrafficMap, wkSvc business.WorkloadService)
 			rate4xx, hasRate4xx := n.Metadata["rate4xx"]
 			rate5xx, hasRate5xx := n.Metadata["rate5xx"]
 			numOutEdges := len(n.Edges)
-			if numOutEdges == 0 && (!hasRate4xx || rate4xx.(float64) == 0) && (!hasRate5xx || rate5xx.(float64) == 0) {
+			if numOutEdges == 0 && (!hasRate4xx || rate4xx.(float64) == 0) && (!hasRate5xx || rate5xx.(float64) == 0) && !isExternalService(n, serviceEntries) {
```

Comment:
If you chose to do the map of external mesh hosts I'd rather see that quick check done first.  if it's an external host just continue, and don't have the check appended to this long conditional.